helm/charts/hpe-csi-driver/values.yaml

```diff
@@ -8,53 +8,82 @@ imagePullPolicy: 'IfNotPresent'
 logLevel: info
 
-## For creating the StorageClass automatically:
-storageClass:
-  create: true
-
-  ## Set StorageClass as the default StorageClass
-  ## Ignored if storageClass.create is false
-  defaultClass: false
+controller:
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  nodeSelector: {}
+    # disktype: ssd
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  tolerations:
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  affinity: {}
+    # nodeAffinity:
+    #   requiredDuringSchedulingIgnoredDuringExecution:
+    #     nodeSelectorTerms:
+    #     - matchExpressions:
+    #       - key: az
+    #         operator: In
+    #         values:
+    #         - az-east
+    #         - az-west
 
-  ## Set a StorageClass name
-  ## Ignored if storageClass.create is false
-  name: hpe-standard
-  # Enable volume resize
-  ## Ignored if storageClass.create is false
-  allowVolumeExpansion: true
-  # Default storage class parameters
-  parameters:
-    fsType: xfs
-    volumeDescription: "Volume created by the HPE CSI Driver for Kubernetes"
-    accessProtocol: "iscsi"
+node:
+  # iSCSI settings
+  iscsi:
+    # Username when CHAP authentication is enabled for iSCSI
+    chapUser: ""
+    # Password when CHAP authentication is enabled for iSCSI
+    chapPassword: ""
+  # For controlling automatic iscsi/multipath package installation (default: false)
+  disableNodeConformance: false
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  nodeSelector: {}
+    # disktype: ssd
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  tolerations: []
+    # - effect: NoSchedule
+    #   key: node-role.kubernetes.io/master
+    #   operator: Exists
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  affinity: {}
+    # nodeAffinity:
+    #   requiredDuringSchedulingIgnoredDuringExecution:
+    #     nodeSelectorTerms:
+    #     - matchExpressions:
+    #       - key: az
+    #         operator: In
+    #         values:
+    #         - az-east
+    #         - az-west
 
-## For controlling automatic iscsi/multipath package installation (default: false)
-disableNodeConformance: false
-
-# CRD creation controls
-## For HELM 3 CRD's are handled automatically without this flag.
-## This flag should be enabled only for HELM 2
-crd:
-  nodeInfo:
-    create: false
-  volumeInfo:
-    create: false
-
-# values for CHAP Authentication
-iscsi:
-  chapUser:
-  chapPassword:
\ No newline at end of file
+csp:
+  servicePort: "8080"
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  nodeSelector: {}
+    # disktype: ssd
+  # These values map directly to yaml in the deployment spec, see the kubernetes docs for info
+  tolerations:
```

Comment:
Yes, a list of tolerations can be added under this field with different key-value pairs and effect. The use case you mentioned can be covered with this.

Reply:
@datamattsson please note that these are not trivial to fetch from UI(or through questions.yaml). So, these will initially be applicable through CLI.

Reply:
@shivamerla sounds reasonable. This is a use case that expands on some of the NFS Server Provisioner feedback I have, where I want to run the NFS servers and RWO PVCs on cordoned and labelled nodes.

Reply:
As NFS deployment is controlled through SC params (map[string]string), it will be tricky. I think we can achieve this for NFS using ConfigMap hpe-nfs-config.

Reply:
I was thinking we can work around it with a simple true/false in the StorageClass or simply add the toleration to the deployment if we find labelled nodes.

Reply:
sure Michael. We can discuss NFS changes further.